folding-schemes/src/folding/circuits/nonnative.rs

```diff
@@ -88,7 +114,12 @@ where
                 &C::BaseField::one(),
                 optimization_type,
             )?;
-        return Ok((x, y));
+        let infinity =
```

Comment:
Regarding the question, they were as `NonNativeFieldVar` because when calling `get_cm_coordinates` it was returning 3 `NonNativeFieldVars` which can then be all together set into an array of `NonNativeFieldVar` (having one of the three as `FpVar` or `Boolean<CF>` would be a bit more tedious. But on the same time the nonnative representation was using extra constraints that were not really needed, and could be avoided as just a `Boolean<CF>` or `FpVar`.
But anyways, we don't actually need it (more details in the other comment: https://github.com/privacy-scaling-explorations/folding-schemes/pull/81#discussion_r1544671766), so just removed the infinity flag for the points representation.